src/EditorFeatures/Core.Wpf/InlineHints/InlineHintsKeyProcessorProvider.cs

```diff
@@ -0,0 +1,140 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable disable
+
+using System;
+using System.ComponentModel.Composition;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Windows.Input;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.InlineHints;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.Utilities;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Editor.InlineHints
+{
+    /// <summary>
+    /// Key processor that allows us to toggle inline hints when a user hits ctrl-alt.
+    /// </summary>
+    [Export(typeof(IKeyProcessorProvider))]
+    [TextViewRole(PredefinedTextViewRoles.Document)]
+    [ContentType(ContentTypeNames.RoslynContentType)]
+    [Name(nameof(InlineHintsKeyProcessorProvider))]
+    [Order(Before = "default")]
+    internal class InlineHintsKeyProcessorProvider : IKeyProcessorProvider
+    {
+        private static readonly ConditionalWeakTable<IWpfTextView, InlineHintsKeyProcessor> s_viewToProcessor = new();
+
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public InlineHintsKeyProcessorProvider()
+        {
+        }
+
+        public KeyProcessor GetAssociatedProcessor(IWpfTextView wpfTextView)
+        {
+            return s_viewToProcessor.GetValue(wpfTextView, v => new InlineHintsKeyProcessor(v));
+        }
+
+        private class InlineHintsKeyProcessor : KeyProcessor
+        {
+            private readonly IWpfTextView _view;
+
+            public InlineHintsKeyProcessor(IWpfTextView view)
+            {
+                _view = view;
+                _view.Closed += OnViewClosed;
+                _view.LostAggregateFocus += OnLostFocus;
+            }
+
+            private static bool IsCtrlOrAlt(KeyEventArgs args)
+                => args.Key is Key.LeftCtrl or Key.RightCtrl or Key.LeftAlt or Key.RightAlt;
+
+            private Workspace GetWorkspace()
+            {
+                var document =
+                    _view.BufferGraph.GetTextBuffers(b => true)
+                                     .Select(b => b.CurrentSnapshot.GetOpenDocumentInCurrentContextWithChanges())
+                                     .WhereNotNull()
+                                     .FirstOrDefault();
+
+                return document?.Project.Solution.Workspace;
+            }
+
+            private void OnViewClosed(object sender, EventArgs e)
```

Comment:
If I hold down Ctrl+Alt, and then close the view, is the option stuck on?

Reply:
Seems to work fine. If you hold it down, switch tabs, then let go, and switch back, you can see the hints for a tiny bit before they go away, but its not a problem at all.

Reply:
i'll turn things off here for paranoia's sake :)